pkg/alertmanager/operator.go

```diff
@@ -469,6 +637,336 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+func (c *Operator) provisionAlertmanagerConfiguration(ctx context.Context, am *monitoringv1.Alertmanager) error {
+	// If no AlertmanagerConfig selectors are configured, the user wants to
+	// manage configuration themselves.
+	if am.Spec.AlertmanagerConfigSelector == nil {
```

Comment:
TODO: in this case we'll still need to copy the base secret (either that or choose to mount the base secret to the pods).